pkg/util/deployment/deployment.go

```diff
@@ -327,21 +335,19 @@ func IsPodAvailable(pod *api.Pod, minReadySeconds int32) bool {
 	return false
 }
 
-func GetPodsForReplicaSets(c clientset.Interface, replicaSets []*extensions.ReplicaSet) ([]api.Pod, error) {
+func GetPodsForReplicaSets(replicaSets []*extensions.ReplicaSet, podList *api.PodList) ([]api.Pod, error) {
 	allPods := map[string]api.Pod{}
 	for _, rs := range replicaSets {
 		if rs != nil {
 			selector, err := unversioned.LabelSelectorAsSelector(rs.Spec.Selector)
 			if err != nil {
 				return nil, fmt.Errorf("invalid label selector: %v", err)
 			}
-			options := api.ListOptions{LabelSelector: selector}
-			podList, err := c.Core().Pods(rs.ObjectMeta.Namespace).List(options)
-			if err != nil {
-				return nil, fmt.Errorf("error listing pods: %v", err)
-			}
 			for _, pod := range podList.Items {
-				allPods[pod.Name] = pod
+				podLabelsSelector := labels.Set(pod.ObjectMeta.Labels)
+				if selector.Matches(podLabelsSelector) {
+					allPods[pod.Name] = pod
```

Comment:
Is it still necessary to use a map for deduping here since the list of Pods is passed in, and presumably does not contain the same Pod multiple times?